File: presentation/src/main/java/com/example/presentation/utils/DateUtils.java
package com.example.presentation.utils;

import android.os.Build;


import org.jetbrains.annotations.NotNull;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.time.YearMonth;
import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;
import java.util.Locale;

/**
 * Утилитный класс
 */
public final class DateUtils {


    private DateUtils() {
    }

    /**
     * Метод для получения отформатированной текущей даты
     *
     * @param currentPosition текущая позиция элемента на экрана, по которй идет вычесление сдвига текущей даты
     * @return отформатированная дата
     */
    @NotNull
    public static String getDateOffset(int currentPosition) {
        Calendar calendar = Calendar.getInstance();
        calendar.add(Calendar.DATE, -currentPosition);
        Date dateWithOffset = calendar.getTime();
        return formattingDate(dateWithOffset);
    }

    /**
     * Для получения количества дней в году
     *
     * @return количество дней в году
     */
    public static int getLengthOfYear() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            return YearMonth.now().lengthOfYear();
        } else {
            Calendar calendar = Calendar.getInstance();
            int day = calendar.get(Calendar.DAY_OF_MONTH);
            int month = calendar.get(Calendar.MONTH);
            int year = calendar.get(Calendar.YEAR);
            Calendar gregorianCalendar = new GregorianCalendar(year, month, day);
            return gregorianCalendar.getActualMaximum(Calendar.DAY_OF_YEAR);
        }
    }

    @NotNull
    private static String formattingDate(@NotNull Date date) {
        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
        return dateFormat.format(date);
    }
}

File: presentation/src/main/java/com/example/presentation/utils/DateUtils.kt
package com.example.presentation.utils

import android.os.Build
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.time.YearMonth
import java.util.*

/**
 * Утилитный класс
 */
object DateUtils {
    /**
     * Метод для получения отформатированной текущей даты
     *
     * @param currentPosition текущая позиция элемента на экрана, по которй идет вычесление сдвига текущей даты
     * @return отформатированная дата
     */
    @JvmStatic
    fun getDateOffset(currentPosition: Int): String {
        val calendar = Calendar.getInstance()
        calendar.add(Calendar.DATE, -currentPosition)
        val dateWithOffset = calendar.time
        return formattingDate(dateWithOffset)
    }

    /**
     * Для получения количества дней в году
     *
     * @return количество дней в году
     */
    @JvmStatic
    fun lengthOfYear(): Int {
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            YearMonth.now().lengthOfYear()
        } else {
            val calendar = Calendar.getInstance()
            val day = calendar[Calendar.DAY_OF_MONTH]
            val month = calendar[Calendar.MONTH]
            val year = calendar[Calendar.YEAR]
            val gregorianCalendar: Calendar = GregorianCalendar(year, month, day)
            gregorianCalendar.getActualMaximum(Calendar.DAY_OF_YEAR)
        }
    }

    @JvmStatic
    private fun formattingDate(date: Date): String {
        val dateFormat: DateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
        return dateFormat.format(date)
    }
}